staging/src/k8s.io/apiserver/pkg/server/mux/pathrecorder.go

```diff
@@ -96,9 +96,13 @@ func NewPathRecorderMux(name string) *PathRecorderMux {
 
 // ListedPaths returns the registered handler exposedPaths.
 func (m *PathRecorderMux) ListedPaths() []string {
+	m.lock.Lock()
+	defer m.lock.Unlock()
+
 	handledPaths := append([]string{}, m.exposedPaths...)
 	sort.Strings(handledPaths)
 
+	m.refreshMuxLocked()
```

Comment:
Not sure why this needs to be called here, this function does not modify the mux

Reply:
Look at the comment: https://github.com/kubernetes/kubernetes/blob/5a21b23b9e1a2501f5f3692545aa5ba2bd0f68fd/staging/src/k8s.io/apiserver/pkg/server/mux/pathrecorder.go#L113-L114

I think it's necessary, also see other calls in this file.

Reply:
A lot of other calls use it to refresh the mux after making a change that affects the mux. This call is read-only and just copies a list. I dont see how that comment means it is necessary to call this function here.

Reply:
okay, your words make sense, It is better that if we can get an ack from other reviewers.

Reply:
Can you take a look?  Do we need to call `m.refreshMuxLocked()` here?

/cc @pohly

Reply:
What Alex says, you lock the access to `exposedPaths` to make a copy on an array and avoid races if someone is writing on it, I can't see why we need to call `m.refreshMuxLocked()`

Reply:
thanks for the comment, I checked the code base, we really don't need to call `m.refreshMuxLocked()`. we need to call `m.refreshMuxLocked()` if we need to read/write `m.pathToHandler` and `m.prefixToHandler`. and I think the `m.trackCallers` function also looks suspicious because it is also read/write `m.pathStacks`.

Reply:
thanks for the comment, I checked the code base, and we don't need to call `m.refreshMuxLocked()`. we need to call `m.refreshMuxLocked()` if we need to read/write `m.pathToHandler` and `m.prefixToHandler`. and I think the `m.trackCallers` function also looks suspicious because it also read/write `m.pathStacks`.